vikunja: 0.22.1 -> 0.23.0

leona-ya opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

https://vikunja.io/blog/2024/02/whats-new-in-vikunja-0.23.0/
https://github.com/go-vikunja/vikunja/blob/main/CHANGELOG.md#0230---2024-02-10

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

leona-ya committed:
vikunja-api, vikunja-frontend: drop

Dropped in favor of a single "vikunja" package that is possible with
the packaging introduced upstream at 0.23.0.

leona-ya committed:
vikunja: init at 0.23.0

https://vikunja.io/blog/2024/02/whats-new-in-vikunja-0.23.0/
https://github.com/go-vikunja/vikunja/blob/main/CHANGELOG.md#0230---2024-02-10

leona-ya committed:
nixos/vikunja: adapt for vikunja 0.23.0

This version removes dedicated api and frontend packages.

leona-ya commented:
Now also the (unit) tests of frontend and backend get executed.

kolaente commented on nixos/modules/services/web-apps/vikunja.nix:
```diff
@@ -124,10 +123,6 @@ in {
     services.nginx.virtualHosts."${cfg.frontendHostname}" = mkIf cfg.setupNginx {
```
Since Vikunja now bundles the frontend with the api, is there any reason to keep the nginx config?

IMHO, people who want to expose Vikunja under a domain or want otherwise customize their hosting experience (which would be a reason to keep nginx) probably have a reverse proxy of sorts set up on their system already and want to integrate with that instead.

leona-ya replied:
Probably right, as it's now very easy to setup.

leona-ya committed:
nixos/vikunja: remove setupNginx option

The required nginx configuration is now really simple, and e.g. SSL/ACME
already required the user to interact with `services.nginx.virtualHosts`.
Therefore, and to reduce complexity, we now leave the web server
configuration to the user.

kolaente reviewed (approved):
Tested locally, LGTM

I'd also add the release blog post: https://vikunja.io/blog/2024/02/whats-new-in-vikunja-0.23.0/

pinpox commented:
Result of `nixpkgs-review pr 289961` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vikunja</li>
  </ul>
</details>
